src/Repository/ProgrammeRepository.php

```diff
@@ -77,20 +88,20 @@ public function getPaginated(int $page, int $limit): array
     public function getPaginatedFilteredSorted(
         array $paginate,
         array $filters,
-        string $sort,
-        string $direction
+        ?string $sort,
+        ?string $direction
     ): array {
         $query = $this->entityManager
             ->createQueryBuilder()
             ->select('p')
             ->from('App\Entity\Programme', 'p')
-            ->setFirstResult(($paginate['currentPage'] * $paginate['maxPerPage']) - $paginate['maxPerPage'])
-            ->setMaxResults($paginate['maxPerPage']);
+            ->setFirstResult($paginate['size'] * ($paginate['page'] - 1))
+            ->setMaxResults($paginate['size']);
 
         foreach ($filters as $key => $value) {
-            if ('' != $value) {
-                $query = $query->where("p.$key = :$key");
-                $query->setParameter(':key', $value);
+            if (null != $value) {
+                $query->where("p.$key = :value");
```

Comment:
Is there a validation of the $key (the filter field) done anywhere? Unless that is done, this code is dangerous.

Reply:
It wasn't, but i added a validation.